[docs] Add note re. running lakectl under Docker (#5353)
treeverse/lakeFS: docs/reference/cli.md

  - ./commands.html
---

# `lakectl` CLI Reference
{:.no_toc}

> `lakectl` is a command-line tool to explore manage and work with lakeFS.

Reviewer: ```suggestion
lakectl is the lakeFS command-line tool.
```

Reviewer 2: I think this is punchier but now confusing: it's the CLI client, but we also run lakefs sometimes.

Reviewer: Fair enough. How about:
```suggestion
`lakectl` is a command-line tool to communicate with the lakeFS server.
```

{% include toc.html h_max=2 %}

> ## Installing `lakectl` locally

Reviewer: ```suggestion
## Installing lakectl locally
```
No need to codify "lakectl". It's the name of the tool, not only the name of the command (inspiration: [kubectl](https://kubernetes.io/docs/reference/kubectl/)).

> `lakectl` is distributed as a single binary, with no external dependencies - and is available for MacOS, Windows and Linux. You can also run it from Docker - see [below](#runninglakectlfromdocker).

Reviewer: ```suggestion
lakectl is available for Linux, MacOS. You can also [run it using Docker](#runninglakectlfromdocker).
```

`lakectl` configuration items can each be controlled by an environment variable. The variable name will have a prefix of
*LAKECTL_*, followed by the name of the configuration, replacing every '.' with a '_'. Example: `LAKECTL_SERVER_ENDPOINT_URL` 
controls `server.endpoint_url`.

## Running lakectl from Docker

If you'd rather run `lakectl` from a Docker container you can do so by passing configuration elements as environment variables. Here is an example: 

```bash
$ docker run --rm --interactive --tty \
          -e LAKECTL_CREDENTIALS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE \
          -e LAKECTL_CREDENTIALS_SECRET_ACCESS_KEY=xxxxx
          -e LAKECTL_SERVER_ENDPOINT_URL=https://host.us-east-2.lakefscloud.io/ \
          --entrypoint lakectl treeverse/lakefs \
          repo list
```

> _Bear in mind that if you are running lakeFS itself locally you will need to account for this in your networking configuration of the Docker container. That is to say, `localhost` to a Docker container is itself, not the host machine on which it is running._

Reviewer: This is an explanation of how networking in Docker works - not really relevant here IMO. If you want to mention it, use an example of how to run locally.

Author: I'm going to leave this in for pragmatic reasons (Docker/local always trips people up), and I've [logged an issue](https://github.com/treeverse/lakeFS/issues/5358) to add example of running it locally.
